src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/RegularExpressionAttribute.cs

```diff
@@ -25,6 +25,18 @@ public RegularExpressionAttribute([StringSyntax(StringSyntaxAttribute.Regex)] st
             MatchTimeoutInMilliseconds = 2000;
         }
 
+        /// <summary>
+        ///     Constructor that accepts the regular expression pattern
+        /// </summary>
+        /// <param name="regex">The regular expression to use.  It cannot be null.</param>
+        public RegularExpressionAttribute(Regex regex)
+            : base(() => SR.RegexAttribute_ValidationError)
+        {
+            Regex = regex;
+            Pattern = regex.ToString();
+            MatchTimeoutInMilliseconds = 2000;
```

Comment:
TODO: Somehow forward MatchTimeoutInMilliseconds to the Regex object; there's a TimeSpan attribute for the timeout, but no setter.